### PR TITLE
Bump vendored dependencies (part 2)

### PR DESCRIPTION
**Pull request by tklauser**
Part 2 of bumping vendored dependencies, following #14572.

Left out the following non-trivial dependencies:

- `google.golang.org/grpc`, `google.golang.org/protobuf`: requires non-trivial API changes
- `go.etcd.io/etcd/clientv3`, conflicts with `google.golang.org/grpc`:
   ```
  go: finding module for package google.golang.org/grpc/naming
  github.com/cilium/cilium/pkg/kvstore imports
  	go.etcd.io/etcd/clientv3 tested by
  	go.etcd.io/etcd/clientv3.test imports
  	github.com/coreos/etcd/integration imports
  	github.com/coreos/etcd/proxy/grpcproxy imports
  	google.golang.org/grpc/naming: module google.golang.org/grpc@latest found (v1.34.1), but does not contain package google.golang.org/grpc/naming```
- `github.com/containernetworking/*`: will update in lockstep with https://github.com/cilium/cilium/blob/0a444b7aa046d7b55f0c7d4cb4ef92bac6247b03/contrib/packaging/docker/Dockerfile.runtime#L53-L57, see https://github.com/cilium/cilium/pull/14606#discussion_r557258876
- `github.com/onsi/ginkgo`: leads to CI failures (e.g. https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/402/consoleFull), needs further investigation

### CODE REVIEW

**Review by errordeveloper — APPROVED** [review]

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

File: go.mod [hunk 1]
```diff
@@ -22,8 +22,8 @@ require (
 	github.com/cilium/ebpf v0.3.0
 	github.com/cilium/ipam v0.0.0-20201106170308-4184bc4bf9d6
 	github.com/cilium/proxy v0.0.0-20201222161048-8bceb17eb027
-	github.com/containernetworking/cni v0.7.1
-	github.com/containernetworking/plugins v0.8.2
+	github.com/containernetworking/cni v0.8.0
+	github.com/containernetworking/plugins v0.9.0
```
Comment (aanm) [inline go.mod]: We might need to split this one into a different PR and put it together with the changes that are required in here as well:
https://github.com/cilium/cilium/blob/0a444b7aa046d7b55f0c7d4cb4ef92bac6247b03/contrib/packaging/docker/Dockerfile.runtime#L53-L57

edit: it's not a requirement for both to be updated at the same time, but it would be nice to have them self contained in a PR.

**Review by aanm — CHANGES_REQUESTED** [review]
The CI must be failing because of ginkgo dependency update

File: go.mod (hunk 1 above)
Comment (tklauser) [reply go.mod]: Good point. For now I've dropped the `github.com/containernetworking/*` and `github.com/ginkgo/*` bumps from this PR and will send separate PRs for these after local testing.

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
